kloppy/domain/services/state_builder/builders/sequence.py

```diff
@@ -31,7 +31,9 @@ def initial_state(self, dataset: EventDataset) -> Sequence:
         return Sequence(sequence_id=0, team=None)
 
     def reduce_before(self, state: Sequence, event: Event) -> Sequence:
-        if isinstance(event, OPEN_SEQUENCE) and state.team != event.team:
+        if isinstance(event, OPEN_SEQUENCE) and (
+            state.team != event.team or event.is_set_piece
```

Comment:
‘if event.get_qualifier_type(SetPieceQualifier):’ should also work. Otherwise we might end up with a lot ‘is_....’ methods on event.